backend/test/core/usecase/ComputeGroupBalanceUseCase.test.ts

```diff
@@ -0,0 +1,66 @@
+import type { GroupRepository } from "src/core/repository/GroupRepository";
+import { describe, test, expect, beforeEach, vi } from "vitest";
+import { GroupRepositoryMock } from "./test-helpers";
+import { ComputeGroupBalanceUseCase } from "src/core/usecase/ComputeGroupBalanceUseCase";
+import { GroupBalance } from "src/core/entity/GroupBalance";
+import { Group } from "src/core/entity/Group";
+import { GroupMember } from "src/core/entity/GroupMember";
+import { Transaction } from "src/core/entity/Transaction";
+
+describe("Compute balance (use case)", () => {
+  let groupRepo: GroupRepository;
+
+  beforeEach(() => {
+    groupRepo = new GroupRepositoryMock();
+  });
+
+  describe("Happy path", () => {
+    test("Given a group already created with two members and two transactions, it should return the balance of the group", async () => {
+      // Arrange
+      const computeBalance = new ComputeGroupBalanceUseCase(groupRepo);
+      const group = new Group(
+        0,
+        "group name",
+        [
+          new GroupMember(1, "group member 1"),
+          new GroupMember(2, "group member 2"),
+        ],
+        [new Transaction(1, 1, [1, 2], 3), new Transaction(2, 1, [2], 1)]
+      );
+      groupRepo.findById = vi.fn().mockResolvedValue(group);
+
+      // Act
+      const result = await computeBalance.execute(0);
+
+      // Assert
+      const expectedGroupBalance = new GroupBalance(
+        0,
+        new Map<number, number>([
```

Comment:
I don't know if you meant it, but this test is an integration test (you test the usecase + the core behaviour)
If you want to test only the usecase (unit test) you have to mock the compute function (you can see what I did in computeReimbursementPlan)
Otherwise, I think it would be better to indicate in some way that it is an integration test